openreview/tools.py

```diff
@@ -360,85 +346,164 @@ def assign(client, paper_number, conference,
-                client.remove_members_from_group(individual_group, user_entity)
-                unassigned_individual_groups.append(individual_group)
-                unassigned_individual_groups = sorted(unassigned_individual_groups, key=lambda x: x.id)
-
-        return unassigned_individual_groups
-    def add_assignment(user, parent_group, unassigned_individual_groups, individual_groups):
-        '''
-        Helper function that adds the given user from the parent group,
-            and to the next empty individual group.
-
-        Prints the results to the console.
-
-        '''
-        assigned_individual_groups = [a for a in individual_groups if user in a.members]
-
-        if user not in parent_group.members:
-            client.add_members_to_group(parent_group, user)
-            print "{:40s} --> {}".format(user.encode('utf-8'), parent_group.id)
-
-        if not assigned_individual_groups:
-            suffix = next_individual_suffix(unassigned_individual_groups, individual_groups, individual_label)
-            anonreviewer_id = '{}/Paper{}/{}'.format(conference, paper_number, suffix)
-            paper_authors = '{}/Paper{}/Authors'.format(conference, paper_number)
-            individual_group = openreview.Group(
-                id = anonreviewer_id,
-                **individual_group_params)
-
-            individual_group.readers.append(anonreviewer_id)
-            individual_group.nonreaders.append(paper_authors)
-            individual_group.signatories.append(anonreviewer_id)
-            individual_group.members.append(user)
-
-            client.post_group(individual_group)
-            print "{:40s} --> {}".format(user.encode('utf-8'), individual_group.id)
-        else:
-            for g in assigned_individual_groups:
-                print "{:40s} === {}".format(user.encode('utf-8'), g.id)
+
+
+def add_assignment(client, paper_number, conference, reviewer,
+                    parent_group_params = {},
+                    individual_group_params = {},
+                    parent_label = 'Reviewers',
+                    individual_label = 'AnonReviewer'):
+
+    '''
+    "individual groups" are groups with a single member;
+        e.g. conference.org/Paper1/AnonReviewer1
+    "parent group" is the group that contains the individual groups;
+        e.g. conference.org/Paper1/Reviewers
+
+    @paper_number: the number of the paper to assign
+    @conference: the ID of the conference being assigned
+    @reviewer: may be an email address or a tilde ID;
+        adds the given user to the parent and individual groups defined by
+        the paper number, conference, and labels
+    @parent_group_params: optional parameter that overrides the default
+    @individual_group_params: optional parameter that overrides the default
+     '''
+
+    parent_group = get_or_create_reviewer_group(client, paper_number, conference, parent_group_params, parent_label)
```

Comment:
it seems these 4 lines are repeated in both functions: add_assignment and remove_assignment. Could we put then together in a common function?

Reply:
may be we can do together in a single function:

- get or create reviewer group
- get individual group

no need to define them separately